webapp/actions/channel_actions.jsx

```diff
@@ -434,3 +434,29 @@ export function getChannelMembersForUserIds(channelId, userIds, success, error)
         }
     );
 }
+
+export function leaveChannel(channelId, isFavorite, success, error) {
```

Comment:
Instead of passing in `isFavorite`, could you switch over to checking if it's a favourite right in the action it self with a call to `ChannelUtils.isFavoriteChannel(channel)`? Might need to pull the channel from the store first since you only have the id, or add a function to check if the channel is favourite based only on id